specs/CookieManagement.md

```diff
@@ -0,0 +1,507 @@
+        Lax = 1,
+        /// Strict SameSite type. The cookie will only be sent along with "same-site" requests.
+        Strict = 2,
+    };
+
+    runtimeclass CoreWebView2
+    {
+        /// Gets the cookie manager object associated with this CoreWebView2.
+        CoreWebView2CookieManager CookieManager { get; };
+
+        // ...
+    }
+
+    /// Creates, adds or updates, gets, or or view the cookies. The changes would
+    /// apply to the context of the user profile. That is, other WebViews under the
+    /// same user profile could be affected.
+    runtimeclass CoreWebView2CookieManager
+    {
+        /// Create a cookie object with a specified name, value, domain, and path.
+        /// One can set other optional properties after cookie creation.
+        /// This only creates a cookie object and it is not added to the cookie
+        /// manager until you call AddOrUpdateCookie.
+        CoreWebView2Cookie CreateCookieWithDetails(String name, String value, String Domain, String Path);
+
+        /// Creates a cookie whose params matches those of the specified cookie.
+        CoreWebView2Cookie CreateCookie(CoreWebView2 cookie);
+
+        /// Gets a list of cookies matching the specific URI.
+        /// If uri is empty string or null, all cookies under the same profile are
+        /// returned.
+        /// You can modify the cookie objects, call
+        /// CoreWebView2CookieManager.AddOrUpdateCookie, and the changes
+        /// will be applied to the webview.
+        Windows.Foundation.IAsyncOperation<IVectorView> GetCookiesAsync(String uri);
+
+        /// Adds or updates a cookie with the given cookie data; may overwrite
+        /// cookies with matching name, domain, and path if they exist.
+        void AddOrUpdateCookie(CoreWebView2Cookie cookie);
+
+        /// Deletes a cookie whose name and domain/path pair
+        /// match those of the specified cookie.
+        void DeleteCookie(CoreWebView2Cookie cookie);
+
+        /// Deletes cookies with matching name and uri.
+        /// Cookie name is required.
+        /// If uri is specified, deletes all cookies with the given name where domain
+        /// and path match provided URI.
+        void DeleteCookies(String name, String uri);
+
+        /// Deletes cookies with matching name and domain/path pair.
+        /// Cookie name is required.
+        /// If domain is specified, deletes only cookies with the exact domain.
+        /// If path is specified, deletes only cookies with the exact path.
+        void DeleteCookiesWithDomainAndPath(String name, String Domain, String Path);
+
+        /// Deletes all cookies under the same profile.
+        /// This could affect other WebViews under the same user profile.
+        void DeleteAllCookies();
+    }
+
+    /// A list of cookie objects.
+    runtimeclass CoreWebView2CookieList : Windows.Foundation.Collections.IVector<CoreWebView2Cookie>

```

Comment:
You shouldn't need this runtimeclass defined at all right? You use IVector<CoreWebView2Cookie> elsewhere in API surface right?